framework/src/controller/channels/ipc_channel.ts

```diff
@@ -162,37 +191,43 @@ export class IPCChannel extends BaseChannel {
 			return handler(action.params) as T;
 		}
 
-		return new Promise((resolve, reject) => {
-			this._rpcClient.call(
-				'invoke',
-				action.toJSONRPCRequest(),
-				(err: JSONRPC.JSONRPCError | undefined, res: JSONRPC.ResponseObjectWithResult<T>) => {
-					if (err) {
-						return reject(err);
-					}
-					return resolve(res.result);
-				},
-			);
+		// When the handler is in other channels
+		return new Promise(resolve => {
+			// Subscribe to the action Id;
+			this._rpcRequestIds.add(action.id as string);
+			this._subSocket.subscribe(action.id as string);
```

Comment:
I don't get it. Subscribing and Unsubscribing for every single `invoke` would be very expensive.

Reply:
ah we don't need this anymore, I was using it when we were trying to do RPC using pub/sub. I removed it now